hikari/impl/voice.py

```diff
@@ -77,22 +77,36 @@ def _check_if_alive(self) -> None:
         if self._is_closing:
             raise errors.ComponentStateConflictError("Component cannot be used while it's closing")
 
-    async def _disconnect(self) -> None:
-        if self._connections:
-            _LOGGER.info("shutting down %s voice connection(s)", len(self._connections))
-            # We rely on the assumption that _on_connection_close will be called here rather than explicitly
-            # emptying self._connections.
-            await asyncio.gather(*(c.disconnect() for c in self._connections.values()))
-
-    async def disconnect(self) -> None:
+    async def disconnect(self, guild: snowflakes.SnowflakeishOr[guilds.PartialGuild]) -> None:
         self._check_if_alive()
-        await self._disconnect()
+        guild_id = snowflakes.Snowflake(guild)
+
+        if guild_id not in self._connections:
+            raise errors.VoiceError("This application doesn't have any active voice connection in this server")
+
+        shard_id = snowflakes.calculate_shard_id(self._app, guild_id)
```

Comment:
Most of this stuff is done in _on_connection_close

Reply:
Removed duplicated logic :+1: